src/lava/magma/core/model/py/connection.py

```diff
@@ -128,6 +143,37 @@ def __init__(self, proc_params: dict) -> None:
         # initialize TraceRandoms and ConnVarRandom
         self._init_randoms()
 
+    def on_var_update(self):
+        self._learning_rule.x1_tau = self.x1_tau[0]
+        self._learning_rule.x1_impulse = self.x1_impulse[0]
+        self._learning_rule.x2_tau = self.x2_tau[0]
+        self._learning_rule.x2_impulse = self.x2_impulse[0]
+
+        self._learning_rule.y1_tau = self.y1_tau[0]
+        self._learning_rule.y1_impulse = self.y1_impulse[0]
+        self._learning_rule.y2_tau = self.y2_tau[0]
+        self._learning_rule.y2_impulse = self.y2_impulse[0]
+        self._learning_rule.y3_tau = self.y3_tau[0]
+        self._learning_rule.y3_impulse = self.y3_impulse[0]
+
+        self._learning_rule.dw_str = self.dw
+        self._learning_rule.dd_str = self.dd
+        self._learning_rule.dt_str = self.dt
+
+        # store impulses and taus in ndarrays with the right shapes
+        self._store_impulses_and_taus()
+
+        # store active traces per dependency from learning_rule in ndarrays
+        # with the right shapes
+        self._build_active_traces_per_dependency()
+
+        # store active traces from learning_rule in ndarrays
+        # with the right shapes
+        self._build_active_traces()
+
+        # generate LearningRuleApplierBitApprox from ProductSeries
+        self._build_learning_rule_appliers()
```

Comment:
I wonder whether the comments are necessary. The important information is contained in the method names. Maybe these could be renamed a bit to be more precise and then the comments could be removed. This would make the code more explanatory.

Reply:
done